libs/ui-lib/lib/ocm/components/clusterWizard/CredentialsDownload.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import {
+  ClusterCredentials,
+  ClustersAPI,
+  ClusterWizardStep,
+  ClusterWizardStepHeader,
+  getApiErrorMessage,
+  handleApiError,
+  KUBECONFIG_INFO_LINK,
+  useAlerts,
+  useTranslation,
+} from '../../../common';
+import { useClusterWizardContext } from './ClusterWizardContext';
+import ClusterWizardFooter from './ClusterWizardFooter';
+import ClusterWizardNavigation from './ClusterWizardNavigation';
+import { WithErrorBoundary } from '../../../common/components/ErrorHandling/WithErrorBoundary';
+import { Cluster, Credentials } from '@openshift-assisted/types/assisted-installer-service';
+import { Alert, AlertVariant, Checkbox, Grid } from '@patternfly/react-core';
+import { downloadFile, getErrorMessage } from '../../../common/utils';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+
+const CredentialsDownload: React.FC<{ cluster: Cluster }> = ({ cluster }) => {
+  const [isChecked, setIsChecked] = React.useState<boolean>(false);
+  const clusterWizardContext = useClusterWizardContext();
+  const [credentials, setCredentials] = React.useState<Credentials>();
+  const [credentialsError, setCredentialsError] = React.useState('');
+  const [isDownloading, setIsDownloading] = React.useState(false);
+  const { addAlert } = useAlerts();
+  const { t } = useTranslation();
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError('');
+      if (!cluster.id) {
+        return;
+      }
+      try {
+        const response = await ClustersAPI.getCredentials(cluster.id);
+        const data = { username: response.data.username, password: response.data.password };
+        setCredentials(data);
+      } catch (err) {
+        setCredentialsError(`Failed to fetch credentials, ${getErrorMessage(err)}`);
+      }
+    };
+    void fetch();
+  }, [cluster]);
+
+  React.useEffect(() => {
+    fetchCredentials();
+  }, [fetchCredentials]);
```

Comment:
no need to split to `useCallback` and `useEffect`. Just put everything into `useEffect`